runtime/libia2/memory_maps.c

```diff
@@ -136,11 +136,11 @@ static void label_memory_map(FILE *log, uintptr_t start_addr) {
   }
 }
 
-#else // IA2_DEBUG_LOG
+#else // IA2_DEBUG_MEMORY
 
-void label_memory_map(FILE *log, uintptr_t start_addr) {}
+static void label_memory_map(FILE *log, uintptr_t start_addr) {}
```

Comment:
Why can't `ia2_log_memory_maps` be defined as the no-op for `!IA2_DEBUG_MEMORY` instead of this function?

Reply:
If it's to support the config `permissive_mode + !IA2_DEBUG_MEMORY` I think it might make sense to require `IA2_DEBUG_MEMORY` to use permissive mode. @fw-immunant any thoughts?

Reply:
Permissive mode is already somewhat invasive, so I don't think it's likely to be a problem for it to require `IA2_DEBUG_MEMORY`. But I don't have a strong opinion either way here.

Reply:
> Why can't `ia2_log_memory_maps` be defined as the no-op for `!IA2_DEBUG_MEMORY` instead of this function?

Without `IA2_DEBUG_MEMORY`, it just prints what's in `/proc/self/maps` (with some columns removed), which it doesn't need to track any memory maps to do, which add overhead, etc.  So I thought it's good to still print that without `IA2_DEBUG_MEMORY`.